src/Subscriptions/Storage/RedisStorageManager.php

```diff
@@ -0,0 +1,169 @@
+ */
+class RedisStorageManager implements StoresSubscriptions
+{
+    /**
+     * @var string
+     */
+    public const TOPIC_KEY = 'graphql.topic';
+
+    /**
+     * @var string
+     */
+    public const SUBSCRIBER_KEY = 'graphql.subscriber';
+
+    /**
+     * @var \Illuminate\Redis\Connections\Connection
+     */
+    private $connection;
+
+    /**
+     * The time to live for items in the cache.
+     *
+     * @var int|null
+     */
+    protected $ttl;
+
+    public function __construct(Repository $config, Factory $redis)
+    {
+        $this->connection = $redis->connection(
+            $config->get('lighthouse.broadcasters.echo.connection', 'default')
+        );
+        $this->ttl = $config->get('lighthouse.subscriptions.storage_ttl', null);
+    }
+
+    public function subscriberByChannel(string $channel): ?Subscriber
+    {
+        return $this->getSubscriber(
+            $this->channelKey($channel)
+        );
+    }
+
+    /**
+     * @return Collection<Subscriber>
+     */
+    public function subscribersByTopic(string $topic): Collection
+    {
+        $subscriberIds = $this->connection->command('smembers', [$this->topicKey($topic)]);
+        $subscriberIds = array_map([$this, 'channelKey'], $subscriberIds);
+        $subscribers = $this->connection->command('mget', [$subscriberIds]);
+
+        return (new Collection($subscribers))
+            ->map(function ($subscriber) {
+                return $this->unserialize($subscriber);
+            })
+            ->filter();
+    }
+
+    public function storeSubscriber(Subscriber $subscriber, string $topic): void
+    {
+        $subscriber->topic = $topic;
+
+        $topicKey = $this->topicKey($topic);
+        $this->connection->command('sadd', [
+            $topicKey,
+            $subscriber->channel,
+        ]);
+        if (isset($this->ttl)) {
+            $this->connection->command('expire', [$topicKey, $this->ttl]);
+        }
+
+        $this->connection->command('set', array_merge([
+            $this->channelKey($subscriber->channel),
+            $this->serialize($subscriber),
+        ], $this->ttl ? ['EX', $this->ttl] : []));
+    }
+
+    /**
+     * @return Subscriber|null
+     */
+    public function deleteSubscriber(string $channel)
+    {
+        $key = $this->channelKey($channel);
+        $subscriber = $this->getSubscriber($key);
+
+        if ($subscriber) {
+            $this->connection->command('del', [$key]);
+            $this->connection->command('srem', [
+                $this->topicKey($subscriber->topic),
+                $channel,
+            ]);
+        }
+
+        return $subscriber;
+    }
+
+    /**
+     * @param string $channelKey
+     */
+    protected function getSubscriber($channelKey): ?Subscriber
+    {
+        $subscriber = $this->unserialize(
+            $this->connection->command('get', [$channelKey])
+        );
+
+        if ($subscriber instanceof Subscriber) {
```

Comment:
What if it is not?

Reply:
Well then it usually is bool `true`.
It's a weird bug which I saw in the tests with an older php and laravel version.
I think there was an issue with `unserialize`.

Reply:
I would like to get to the root cause of this bug. If it is not a `Subscriber` or `null`, something is wrong.